drake/examples/kuka_iiwa_arm/kuka_control.cc

```diff
@@ -0,0 +1,180 @@
+    iiwa_status_.utime = cur_time_us;
+    robot_controller_reference_.utime = cur_time_us;
+
+    lcmt_iiwa_command iiwa_command;
+    iiwa_command.num_joints = kNumJoints;
+    iiwa_command.joint_position.resize(kNumJoints, 0.);
+    iiwa_command.num_torques = kNumJoints;
+    iiwa_command.joint_torque.resize(kNumJoints, 0.);
+
+    Eigen::VectorXd joint_position_desired(kNumJoints); 
+    Eigen::VectorXd joint_velocity_desired(kNumJoints); 
+    Eigen::VectorXd joint_accel_desired(kNumJoints); 
+
+    bool half_servo_rate_flag_ = true; // make the iiwa command get published every two servo loops
+
+    while (true) {      
+      // Call lcm handle until at least one message is processed
+      while (0 == lcm_.handleTimeout(10)) { }
+      DRAKE_ASSERT(iiwa_status_.utime != -1);
+      DRAKE_ASSERT(robot_controller_reference_.utime != -1);
+      cur_time_us = iiwa_status_.utime;
+
+      if (controller_trigger_) {  
+        const int kNumDof = 7;
+        iiwa_command.utime = iiwa_status_.utime;
+
+        // Kuka-Controller (Currently implement an inverse dynamics controller)
+        // Set desired joint position, velocity and acceleration
+        for (int joint = 0; joint < kNumDof; joint++){
+          joint_position_desired(joint) = robot_controller_reference_.joint_position_desired[joint];
+          joint_velocity_desired(joint) = robot_controller_reference_.joint_velocity_desired[joint];
+          joint_accel_desired(joint) = robot_controller_reference_.joint_accel_desired[joint];
+        }
+
+        double *qptr = &iiwa_status_.joint_position_measured[0];
+        Eigen::Map<Eigen::VectorXd> q(qptr, kNumDof);
+        double *qdptr = &iiwa_status_.joint_velocity_estimated[0];
+        Eigen::Map<Eigen::VectorXd> qd(qdptr, kNumDof);
+
+        // Computing inverse dynamics torque command
+        KinematicsCache<double> cache = tree_.doKinematics(q, qd);
+        const RigidBodyTree<double>::BodyToWrenchMap no_external_wrenches;
+
+        // note that, the gravity term in the inverse dynamics is set to zero.
+        Eigen::VectorXd torque_command = tree_.inverseDynamics(cache, no_external_wrenches, joint_accel_desired, false);
+        
+        Eigen::VectorXd z = Eigen::VectorXd::Zero(kNumDof); 
+        KinematicsCache<double> cache0 = tree_.doKinematics(q, qd);
+        Eigen::VectorXd gravity_torque = tree_.inverseDynamics(cache0, no_external_wrenches, z, false);
+        torque_command -= gravity_torque;
+
+        // PD position control
+        Eigen::VectorXd position_ctrl_torque_command(kNumDof);
+        Eigen::VectorXd Kp_pos_ctrl(kNumDof); // 7 joints
+        // Kp_pos_ctrl << 225, 289, 144, 49, 324, 49, 49;
+        Kp_pos_ctrl << 100, 100, 100, 100, 100, 50, 50;
+        Eigen::VectorXd Kd_pos_ctrl(kNumDof); // 7 joints
+        // Kd_pos_ctrl << 30, 34, 24, 14, 36, 14, 14;
+        Kd_pos_ctrl << 19, 19, 19, 19, 19, 14, 14;
+        // (TODOs) Add integral control (anti-windup)
```

Comment:
Thank you. So far, no vibration shows up. The motion I mentioned today is kind of one or two overshoots at the end of the motion (the end-effector slightly swings back and forth instead of stopping exactly where it is supposed to be). I will tune the Kd gains.